java/densify-and-generalize/src/main/java/com/esri/arcgisruntime/sample/densifygeneralize/MainActivity.java

```diff
@@ -0,0 +1,203 @@
+
+    // show the result (densified and generalized) point as magenta dots on the map
+    mResultPointGraphic = new Graphic();
+    mResultPointGraphic.setSymbol(new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.CIRCLE, 0xFFFF00FF, 7));
+    graphicsOverlay.getGraphics().add(mResultPointGraphic);
+
+    // connect the results points with a magenta line
+    mResultPolylineGraphic = new Graphic();
+    mResultPolylineGraphic.setSymbol(new SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, 0xFFFF00FF, 3));
+    graphicsOverlay.getGraphics().add(mResultPolylineGraphic);
+
+    mMapView.setViewpointGeometryAsync(mOriginalPolyline.getExtent(), 100);
+
+    // set defaults
+    mMaxDeviationSlider.setMax(249);
+    mMaxSegmentLengthSlider.setMax(400);
+    mGeneralizeCheckBox.setChecked(true);
+    mDensifyCheckBox.setChecked(true);
+    setListeners();
+    updateGeometry();
+  }
+
+  /**
+   * Adds listeners to both seek bars and both check boxes.
+   */
+  private void setListeners() {
+    mMaxDeviationSlider.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+      @Override public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+        updateGeometry();
+      }
+
+      @Override public void onStartTrackingTouch(SeekBar seekBar) {
+      }
+
+      @Override public void onStopTrackingTouch(SeekBar seekBar) {
+      }
+    });
+
+    mMaxSegmentLengthSlider.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+      @Override public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+        updateGeometry();
+      }
+
+      @Override public void onStartTrackingTouch(SeekBar seekBar) {
+      }
+
+      @Override public void onStopTrackingTouch(SeekBar seekBar) {
+      }
+    });
+
+    mGeneralizeCheckBox.setOnClickListener(v -> updateGeometry());
+    mDensifyCheckBox.setOnClickListener(v -> updateGeometry());
+
+  }
+
+  /**
+   * Called when any of the densify/generalize option values are changed. Applies the densify and generalize
+   * operations to the original polyline and updates the result graphics with the result geometry.
+   */
+  private void updateGeometry() {
+    Polyline tempPolyline = mOriginalPolyline;
+    if (mGeneralizeCheckBox.isChecked()) {
+      tempPolyline = (Polyline) GeometryEngine.generalize(tempPolyline, mMaxDeviationSlider.getProgress() + 1,
+          true);
+    }
+    if (mDensifyCheckBox.isChecked()) {
+      tempPolyline = (Polyline) GeometryEngine.densify(tempPolyline, mMaxSegmentLengthSlider.getProgress() + 100);
+    }
+    mResultPolylineGraphic.setGeometry(tempPolyline);
+    Multipoint multipoint = new Multipoint(tempPolyline.getParts().getPartsAsPoints());
+    mResultPointGraphic.setGeometry(multipoint);
+  }
+
+  /**
+   * Creates a collection of points along the Willamette River in Portland, OR.
+   *
+   * @return points
+   */
+  private static PointCollection createShipPoints() {
```

Comment:
Why are these called Ship points? I wouldn't want to sail a ship on a generalized line...

Anyways, it looks like all the other platforms have already called it that. I guess the 'ship has sailed' on this :P

Reply:
I agree that the two checkboxes don't really do much to the sample. I was following what Java did. Should I remove them ?

Reply:
Lets leave them in if they're implemented in other samples.